### PR TITLE
Parametric Point2Point2Range

### PR DESCRIPTION
**Pull request by Affie**
cc @dehann  
PR for discussing what to do with factors like Point2Point2Range.
See `EuclidDistance` like factor and test for batch parametric solve

### CODE REVIEW

File: src/factors/Range2D.jl [hunk 1]
```diff
@@ -17,11 +17,14 @@ function (cfo::CalcFactor{<:Point2Point2Range})(rho, theta, xi, lm)
...
-  #TODO JT - Should this have a sqrt for parametric?
-  # return XX^2 + YY^2 
   return sqrt(XX^2 + YY^2)
 end
 
+#Parametric only function fro Point2Point2Range 
+function (cfo::CalcFactor{<:Point2Point2Range})(rho, xi, lm)
+  # Basically `EuclidDistance`
+  return [rho[1] - norm(lm .- xi)]
+end
```
Comment (dehann) [inline src/factors/Range2D.jl:27]: right, understand now.  Best solution would be to drop the additional `theta` variable, but we should look at how best to do together with the Manifolds.jl upgrade -- lots of overlap.

**Comment by dehann** [comment]
Hi @Affie , with IIF v0.21 tagged tests starting to pass here again although "This and Masters" is failing in parametric.  Perhaps just check that out when you get a chance please.

File: src/factors/Range2D.jl (hunk 1 above)
Comment (dehann) [reply src/factors/Range2D.jl:27]: I never liked the additional `theta` thing.  Perhaps the solution is the same as JuliaRobotics/IncrementalInference.jl#1051

**Comment by Affie** [comment]
Point2Point2Range test is failing because its not converging correctly every time, it depends on the initial values.  It seems like such a small example that I’m sure fine tuning the solver parameters can make it better. 
I mainly opened PR for discussion.  If we are looking at dropping theta, then i would say we keep both definitions until high entropy version works.  In that case i can look at getting this passing an merge.

**Comment by dehann** [comment]
Would making this similar to Roots fix the problem?

https://github.com/JuliaRobotics/IncrementalInference.jl/blob/c6a845763448af40983755449e245d2a7e8b41dc/src/NumericalCalculations.jl#L18

**Comment by Affie** [comment]
We can now consolidate and remove extra `theta` variable https://github.com/JuliaRobotics/IncrementalInference.jl/issues/1051 
I'll update for next release.

**Comment by Affie** [comment]
Combined with #416

**Pull request closed without merging**